### ISSUE TITLE
refactor: Replace @pyroscope/flamegraph dependency

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/jaeger-ui#1870]
The company behind this module has been acquired by Grafana. The repo moved there https://github.com/grafana/pyroscope, and they changed the license to AGPL, to match the rest of Grafana. AGPL is not allowed in [CNCF projects](https://github.com/cncf/foundation/blob/main/allowed-third-party-license-policy.md).

We're currently on an old version, and no new versions are expected. It's not an issue right now, but eventually there may be security issues, if not in that module than in its dependencies.

**Comment by priyanshu-kun** [comment]
Hey Yuri, Any alternative of this @pyroscope/flamegraph package?

**Comment by RipulHandoo** [comment]
Hi @yurishkuro,

I noticed the use of Pyroscope in our codebase [here](https://github.com/jaegertracing/jaeger-ui/blob/main/packages/jaeger-ui/src/components/TracePage/TraceFlamegraph/index.test.js#L23) and the corresponding test case [here](https://github.com/jaegertracing/jaeger-ui/blob/main/packages/jaeger-ui/src/components/TracePage/TraceFlamegraph/testTrace.json#L34-L52).

Considering Pyroscope's move to Grafana with AGPL licensing, I suggest exploring [pprof](https://github.com/google/pprof) as a potential replacement for our Jaeger UI project.

Steps:
1. Research pprof's suitability.
2. Discuss with maintainers.
3. Decide and implement changes.

Ensuring project security and stability is crucial, and I believe pprof could align well with our needs.

**Comment by yurishkuro** [comment]
pprof is a standalone tool, not a viz library.

**Comment by EshaanAgg** [comment]
[`react-flame-graph`](https://github.com/bvaughn/react-flame-graph) seems to be the most popular flame-graph visualisation library out there. It is developed by Brian Vaughn, [a core React developer who worked on introducing the React Profiler](https://legacy.reactjs.org/blog/2018/09/10/introducing-the-react-profiler.html)

It seems like a great library that can be incorporated, but the only caveat being it was last updated 4 years ago.

**Comment by ghost** [comment]
@yurishkuro I suggest evaluating Parca UI's components. Parca defaults to Icicle graphs (root at the top). While functionally identical to flame graphs, it is a visual shift from the current bottom-up pyroscope view.
https://www.parca.dev/docs/icicle-graph-interpretation/

**Comment by yurishkuro** [comment]
> [react-flame-graph](https://github.com/bvaughn/react-flame-graph)

I checked and that package has not been updated in 6-8 years, and has only worked with React 16.

**Comment by ghost** [comment]
Yes, I looked into few other available options and most are unmaintained. Parca UI appears to be the only actively maintained choice. Although it use icicle graphs, but i think it could still be a viable candidate. What do you think?
